presto-main/src/test/java/com/facebook/presto/sql/planner/TestPredicatePushdown.java

```diff
@@ -423,14 +424,24 @@ public void testNonDeterministicPredicateNotPushedDown()
                                                                 ImmutableMap.of("CUST_KEY", "custkey"))))))));
     }
 
+    @Override
+    protected void assertPlan(String sql, PlanMatchPattern pattern)
+    {
+        assertPlan(sql, LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED, pattern, planOptimizer -> !(planOptimizer instanceof StatsRecordingPlanOptimizer) ||
```

Comment:
Add a comment to explain why you are doing this. Add a "TODO" to remove this function once all pushdownPredicate use cases are migrated to the RowExpr version.